src/main/java/org/hisp/dhis/BaseDhis2.java

```diff
@@ -791,6 +871,12 @@ protected <T> T getObjectFromUrl(URI url, Class<T> type) {
 
       log("Response body: '{}'", responseBody);
 
+      if (type == String.class) {
```

Comment:
I think a String is considerd valid JSON / as a JSON "primitive". So wondering if we actually need this if block?